### PR TITLE
Sometimes things need fixin

### PR DESCRIPTION
**Pull request by kbacha**
### Is there a GIF that reflects how this work made you feel?
burgertime

Goes through and resolves a couple of failing specs. There is still one more failing one with the timeout. Will be looking at that soon.

also fixes cargo fmt.

### CODE REVIEW

File: client/src/endpoint/trade.rs [hunk 1]
```diff
@@ -382,6 +382,36 @@ pub struct Aggregations {
...
+impl From<SegmentResolution> for u64 {
+    fn from(res: SegmentResolution) -> u64 {
+        match res {
+            SegmentResolution::OneMin => 60000,
+            SegmentResolution::FiveMin => 300000,
+            SegmentResolution::FifteenMin => 900000,
+            SegmentResolution::OneHour => 3600000,
+            SegmentResolution::OneDay => 86400000,
+            SegmentResolution::OneWeek => 604800000,
+        }
```
Comment (correlator) [inline client/src/endpoint/trade.rs]: 🔥

File: client/src/endpoint/trade.rs [hunk 2]
```diff
@@ -382,6 +382,36 @@ pub struct Aggregations {
...
+    OneMin,
+    /// Represents a five minute segment.
+    FiveMin,
+    /// Represents a fifteen minute segment.
+    FifteenMin,
+    /// Represents an hour long segment.
+    OneHour,
+    /// Represents a day long segment.
+    OneDay,
+    /// Respresents a week long segment.
```
Comment (correlator) [inline client/src/endpoint/trade.rs]: These comments feel a bit superfluous.

File: cli/src/main.rs [hunk 3]
```diff
@@ -455,7 +455,8 @@ fn build_app<'a, 'b>() -> App<'a, 'b> {
                                 .long("resolution")
                                 .takes_value(true)
                                 .required(true)
-                                .help("Segment duration in format <number><unit> where units are s, m, h, d.  ie: 10h == 10 hours"),
+                                .possible_values(&["1m", "5m", "15m", "1h", "1d", "1w"])
+                                .help("The resolution of each segment. Limited to a set of valid inputs")
```
Comment (correlator) [inline cli/src/main.rs]: very cool

File: client/src/endpoint/trade.rs (hunk 2 above)
Comment (kbacha) [reply client/src/endpoint/trade.rs]: Yes, but required for `#[deny(missing_doc)]`

File: client/Cargo.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "stellar-client"
-version = "0.1.1"
+version = "0.1.2"
```
Comment (correlator) [inline client/Cargo.toml]: 💪

**Pull request merged**